src/highlight/HighlightAnnotations.tsx

```diff
@@ -37,6 +39,8 @@ const HighlightAnnotations = (props: Props): JSX.Element => {
         annotations = [],
         createHighlight = noop,
         isCreating = false,
+        isPromoting = false,
```

Comment:
I kept it separate due to the `useEffect` hook on L113 which watches for `isCreating=true` which is interpreted as "explicit create highlight mode" and `selection` as truthy such that it will trigger the current selection to be a staged highlight annotation. But let me see if I can tweak the `handlePromote` to follow the same flow

Reply:
Turns out we need to keep them separate for the hiding/showing of the `PopupHighlight`. We'll need to take a look at improving the creating status as you alluded to